sdk/core/azure-core-management/src/main/java/com/azure/core/management/implementation/AzureProxy.java

```diff
@@ -1,421 +0,0 @@
-     * @return the default HttpPipeline.
-     */
-    public static HttpPipeline createDefaultPipeline(Class<?> swaggerInterface) {
-        return createDefaultPipeline(swaggerInterface, (HttpPipelinePolicy) null);
-    }
-
-    /**
-     * Create the default HttpPipeline.
-     *
-     * @param swaggerInterface The interface that the pipeline will use to generate a user-agent string.
-     * @param credentials The credentials to use to apply authentication to the pipeline.
-     * @return the default HttpPipeline.
-     */
-    public static HttpPipeline createDefaultPipeline(Class<?> swaggerInterface, TokenCredential credentials) {
-        return createDefaultPipeline(swaggerInterface, new BearerTokenAuthenticationPolicy(credentials));
-    }
-
-    /**
-     * Create the default HttpPipeline.
-     *
-     * @param swaggerInterface The interface that the pipeline will use to generate a user-agent string.
-     * @param credentialsPolicy The credentials policy factory to use to apply authentication to the pipeline.
-     * @return the default HttpPipeline.
-     */
-    public static HttpPipeline createDefaultPipeline(Class<?> swaggerInterface, HttpPipelinePolicy credentialsPolicy) {
-        // Order in which policies applied will be the order in which they appear in the array
-        //
-        List<HttpPipelinePolicy> policies = new ArrayList<>();
-        policies.add(new UserAgentPolicy(getDefaultUserAgentString(swaggerInterface)));
-        policies.add(new RetryPolicy());
-        policies.add(new CookiePolicy());
-        if (credentialsPolicy != null) {
-            policies.add(credentialsPolicy);
-        }
-
-        return new HttpPipelineBuilder()
-            .policies(policies.toArray(new HttpPipelinePolicy[0]))
-            .build();
-    }
-
-    /**
-     * Create a proxy implementation of the provided Swagger interface.
-     *
-     * @param swaggerInterface The Swagger interface to provide a proxy implementation for.
-     * @param azureEnvironment The azure environment that the proxy implementation will target.
-     * @param httpPipeline The HTTP httpPipeline will be used to make REST calls. //* @param serializer The serializer
-     * that will be used to convert POJOs to and from request and response bodies.
-     * @param <A> The type of the Swagger interface.
-     * @return A proxy implementation of the provided Swagger interface.
-     */
-    @SuppressWarnings("unchecked")
-    public static <A> A create(Class<A> swaggerInterface, AzureEnvironment azureEnvironment,
-        HttpPipeline httpPipeline) {
-        String baseUrl = null;
-
-        if (azureEnvironment != null) {
-            final AzureHost azureHost = swaggerInterface.getAnnotation(AzureHost.class);
-            if (azureHost != null) {
-                baseUrl = azureEnvironment.url(azureHost.endpoint());
-            }
-        }
-
-        final SwaggerInterfaceParser interfaceParser =
```

Comment:
This is another convenient that the AzureProxy provide for host URL. Since the other libraries cannot access the SwaggerInterfaceParser, each RestService method have to add the @HostParam("url") parameter, which is not developer friendly on the service client side.